libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/scanners/WifiScanner.java

```diff
@@ -153,13 +156,11 @@ synchronized void activatePeriodicScan(final ActiveOrPassiveStumbling stumblingM
         // Ensure that we are constantly scanning for new access points.
         mWifiScanTimer = new Timer();
         mWifiScanTimer.schedule(new TimerTask() {
-            int mPassiveScanCount;
 
             @Override
             public void run() {
                 if (stumblingMode == ActiveOrPassiveStumbling.PASSIVE_STUMBLING &&
-                        mPassiveScanCount++ > AppGlobals.PASSIVE_MODE_MAX_SCANS_PER_GPS) {
-                    mPassiveScanCount = 0;
```

Comment:
Aside: This was unnecessary in the original code. A new instance of this TimerTask would be created after a stop().